homeassistant/components/device_tracker/cppm_tracker.py

```diff
@@ -0,0 +1,85 @@
+"""
+Support for ClearPass Policy Manager.
+
+Allows tracking devices with CPPM.
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    PLATFORM_SCHEMA, DeviceScanner, DOMAIN
+)
+from homeassistant.const import (
+    CONF_HOST, CONF_API_KEY
+)
+
+REQUIREMENTS = ['clearpasspy==1.0.2']
+
+SCAN_INTERVAL = timedelta(seconds=120)
+
+CLIENT_ID = 'client_id'
+
+GRANT_TYPE = 'client_credentials'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CLIENT_ID): cv.string,
+    vol.Required(CONF_API_KEY): cv.string,
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_scanner(hass, config):
+    """Initialize Scanner."""
+    from clearpasspy import ClearPass
+    data = {
+        'server': config[DOMAIN][CONF_HOST],
+        'grant_type': GRANT_TYPE,
+        'secret': config[DOMAIN][CONF_API_KEY],
+        'client': config[DOMAIN][CLIENT_ID]
+    }
+    cppm = ClearPass(data)
+    if cppm.access_token is not None:
+        new_scan = CPPMDeviceScanner(config[DOMAIN], cppm)
+        return new_scan
+    return None
+
+
+class CPPMDeviceScanner(DeviceScanner):
+    """Initialize class."""
+
+    def __init__(self, config, cppm):
+        """Initialize class."""
+        self._cppm = cppm
+        self.results = None
+
+    def scan_devices(self):
+        """Initialize scanner."""
+        self.get_cppm_data()
+        return [device['mac'] for device in self.results]
+
+    def get_device_name(self, device):
+        """Retrieve device name."""
+        return [device['name'] for device in self.results]
+
+    @Throttle(SCAN_INTERVAL)
+    def get_cppm_data(self):
+        """Retrieve data from Aruba Clearpass and return parsed result."""
+        _LOGGER.debug("Access Token: %s", self._cppm.access_token)
```

Comment:
We shouldn't log credentials.

Reply:
I was curious about this rule since the access token is temporary for the instance. The secret stored in the config is really what needs protected. 

Please advise.

Reply:
Ok. What's the benefit of logging the access token?

Reply:
I immediately know the API has connected successfully. I could log a status code, I suppose, in get_scanner. I'll do that.